cmake/kokkos_enable_options.cmake

```diff
@@ -65,6 +65,13 @@ KOKKOS_ENABLE_OPTION(HIP_MULTIPLE_KERNEL_INSTANTIATIONS OFF "Whether multiple ke
 KOKKOS_ENABLE_OPTION(IMPL_DESUL_ATOMICS   ON  "Whether to use desul based atomics - option only during beta")
 KOKKOS_ENABLE_OPTION(DESUL_ATOMICS_EXTERNAL OFF "Whether to use an external desul installation")
 
+KOKKOS_ENABLE_OPTION(IMPL_MDSPAN OFF "Whether to enable experimental mdspan support")
+KOKKOS_ENABLE_OPTION(MDSPAN_EXTERNAL OFF BOOL "Whether to use an external version of mdspan")
+KOKKOS_ENABLE_OPTION(MDSPAN_INTERNAL OFF BOOL "Whether to use an internal version of mdspan even if the compiler supports mdspan")
+mark_as_advanced(Kokkos_ENABLE_IMPL_MDSPAN)
+mark_as_advanced(Kokkos_ENABLE_MDSPAN_EXTERNAL)
+mark_as_advanced(Kokkos_ENABLE_MDSPAN_INTERNAL)
```

Comment:
What combination of these two are valid?

Reply:
To be clear, that is the main thing I want to clarify on this PR.  I can live with everything else.

Reply:
I think they should be mutually exclusive, but both of them could be off at the same time if the compiler has mdspan, and that should be the default if we know the compiler has a working version.

Reply:
OK a rename actually can fix that, also make it impl: Kokkos_ENABLE_IMPL_SKIP_COMPILER_MDSPAN

Rational: we want people to be able to use external mdspan if they need to so that is public, but if they don't do that we do NOT really want to allow them long term to choose between an internal and the compiler version. Its something we rather make the decision for them.

Reply:
My view on this is that we need it to make progress with the inclusion of `mdspan` and so we should force using an internal version of mdspan for now. We can worry about other options later.

Reply:
Renaming to `Kokkos_ENABLE_IMPL_SKIP_COMPILER_MDSPAN`